src/mame/sega/m1comm.cpp

```diff
@@ -619,22 +628,17 @@ void m1comm_device::send_frame(int dataSize){
 	if (!m_line_tx)
 		return;
 
-	std::error_condition filerr;
-	std::uint32_t written;
-
-	filerr = m_line_tx->write(&m_buffer0, 0, dataSize, written);
+	uint32_t written = 0;
+	std::error_condition filerr = m_line_tx->write(&m_buffer0, 0, dataSize, written);
 	if (filerr)
 	{
+		osd_printf_verbose("M1COMM: tx connection error\n");
```

Comment:
Since you’ve got a non-blocking socket, there’s a very real possibility that you’ll end up with a partial write here.  You should be checking `written`.  Blocking really isn’t an option, as you’ll hold up the emulation.  Have you thought about what to do in that situation?  (An asynchronous signal can also cause a partial write to happen.)

Also remember that you won’t get an error from a partial write, you need to attempt to complete the write to see the error.  That’s the reason for those read/write loop helpers in src/lib/util/ioprocs.cpp

Reply:
Are you sure we are using non-blocking sockets here?

As far as posixsocket.cpp and winsocket.cpp go, sockets are opened with TCP_NODELAY (which disables nagle's algorythm)

posixsocket.cpp sets O_NONBLOCK for "domains" only

Reply:
should be fine now.